pkg/kernel/kernel-config/kernel_config-5.10.x-preempt-rt-x86_64.patch

```diff
@@ -0,0 +1,195 @@
+ 
+ #
+ # Power management and ACPI options
+@@ -689,8 +696,6 @@
+ CONFIG_KEXEC_CORE=y
+ CONFIG_HOTPLUG_SMT=y
+ CONFIG_GENERIC_ENTRY=y
+-CONFIG_OPROFILE=y
+-# CONFIG_OPROFILE_EVENT_MULTIPLEX is not set
+ CONFIG_HAVE_OPROFILE=y
+ CONFIG_OPROFILE_NMI_TIMER=y
+ CONFIG_KPROBES=y
+@@ -735,6 +740,7 @@
+ CONFIG_MMU_GATHER_TABLE_FREE=y
+ CONFIG_MMU_GATHER_RCU_TABLE_FREE=y
+ CONFIG_ARCH_HAVE_NMI_SAFE_CMPXCHG=y
++CONFIG_HAVE_ALIGNED_STRUCT_PAGE=y
+ CONFIG_HAVE_CMPXCHG_LOCAL=y
+ CONFIG_HAVE_CMPXCHG_DOUBLE=y
+ CONFIG_ARCH_WANT_COMPAT_IPC_PARSE_VERSION=y
+@@ -771,6 +777,7 @@
+ CONFIG_OLD_SIGSUSPEND3=y
+ CONFIG_COMPAT_OLD_SIGACTION=y
+ CONFIG_COMPAT_32BIT_TIME=y
++CONFIG_ARCH_SUPPORTS_RT=y
+ CONFIG_HAVE_ARCH_VMAP_STACK=y
+ CONFIG_VMAP_STACK=y
+ CONFIG_ARCH_HAS_STRICT_KERNEL_RWX=y
+@@ -872,11 +879,6 @@
+ 
+ CONFIG_PREEMPT_NOTIFIERS=y
+ CONFIG_ASN1=y
+-CONFIG_INLINE_SPIN_UNLOCK_IRQ=y
+-CONFIG_INLINE_READ_UNLOCK=y
+-CONFIG_INLINE_READ_UNLOCK_IRQ=y
+-CONFIG_INLINE_WRITE_UNLOCK=y
+-CONFIG_INLINE_WRITE_UNLOCK_IRQ=y
+ CONFIG_ARCH_SUPPORTS_ATOMIC_RMW=y
+ CONFIG_MUTEX_SPIN_ON_OWNER=y
+ CONFIG_RWSEM_SPIN_ON_OWNER=y
+@@ -884,7 +886,6 @@
+ CONFIG_ARCH_USE_QUEUED_SPINLOCKS=y
+ CONFIG_QUEUED_SPINLOCKS=y
+ CONFIG_ARCH_USE_QUEUED_RWLOCKS=y
+-CONFIG_QUEUED_RWLOCKS=y
+ CONFIG_ARCH_HAS_NON_OVERLAPPING_ADDRESS_SPACE=y
+ CONFIG_ARCH_HAS_SYNC_CORE_BEFORE_USERMODE=y
+ CONFIG_ARCH_HAS_SYSCALL_WRAPPER=y
+@@ -931,9 +932,6 @@
+ CONFIG_MMU_NOTIFIER=y
+ CONFIG_KSM=y
+ CONFIG_DEFAULT_MMAP_MIN_ADDR=65536
+-CONFIG_TRANSPARENT_HUGEPAGE=y
+-CONFIG_TRANSPARENT_HUGEPAGE_ALWAYS=y
+-# CONFIG_TRANSPARENT_HUGEPAGE_MADVISE is not set
+ CONFIG_ARCH_WANTS_THP_SWAP=y
+ # CONFIG_CLEANCACHE is not set
+ # CONFIG_CMA is not set
+@@ -950,7 +948,6 @@
+ CONFIG_ARCH_HAS_PKEYS=y
+ # CONFIG_PERCPU_STATS is not set
+ # CONFIG_GUP_BENCHMARK is not set
+-# CONFIG_READ_ONLY_THP_FOR_FS is not set
+ CONFIG_ARCH_HAS_PTE_SPECIAL=y
+ # end of Memory Management options
+ 
+@@ -1546,7 +1543,6 @@
+ CONFIG_XPS=y
+ CONFIG_CGROUP_NET_PRIO=y
+ CONFIG_CGROUP_NET_CLASSID=y
+-CONFIG_NET_RX_BUSY_POLL=y
+ CONFIG_BQL=y
+ CONFIG_BPF_JIT=y
+ CONFIG_NET_FLOW_LIMIT=y
+@@ -4077,7 +4073,6 @@
+ # CONFIG_LEDS_TRIGGER_DISK is not set
+ CONFIG_LEDS_TRIGGER_HEARTBEAT=m
+ CONFIG_LEDS_TRIGGER_BACKLIGHT=m
+-CONFIG_LEDS_TRIGGER_CPU=y
+ # CONFIG_LEDS_TRIGGER_ACTIVITY is not set
+ CONFIG_LEDS_TRIGGER_GPIO=m
+ CONFIG_LEDS_TRIGGER_DEFAULT_ON=m
+@@ -5692,7 +5687,6 @@
+ CONFIG_TEXTSEARCH_BM=y
+ CONFIG_TEXTSEARCH_FSM=y
+ CONFIG_INTERVAL_TREE=y
+-CONFIG_XARRAY_MULTI=y
+ CONFIG_ASSOCIATIVE_ARRAY=y
+ CONFIG_HAS_IOMEM=y
+ CONFIG_HAS_IOPORT_MAP=y
+@@ -5817,7 +5811,7 @@
+ CONFIG_GENERIC_PTDUMP=y
+ # CONFIG_PTDUMP_DEBUGFS is not set
+ # CONFIG_DEBUG_OBJECTS is not set
+-# CONFIG_DEBUG_SLAB is not set
++# CONFIG_SLUB_STATS is not set
+ CONFIG_HAVE_DEBUG_KMEMLEAK=y
+ # CONFIG_DEBUG_KMEMLEAK is not set
+ # CONFIG_DEBUG_STACK_USAGE is not set
+@@ -5870,6 +5864,7 @@
+ # end of Scheduler Debugging
+ 
+ # CONFIG_DEBUG_TIMEKEEPING is not set
++CONFIG_DEBUG_PREEMPT=y
```

Comment:
Enable DEBUG here will increase overhead and affect the performance, I don't think we should enabled by default...

Reply:
Left as is intentionally. Want to catch any complains from kernel if preempted in unsafe way. Want to see the whole virtualization stack in action. Will disable later.